tests/console/snapper_cleanup.pm

```diff
@@ -73,17 +73,53 @@ sub run {
         assert_script_run "snapper set-config NUMBER_LIMIT=2-10 NUMBER_LIMIT_IMPORTANT=4-10 SPACE_LIMIT=0.5";
     }
 
+    my ($n_scratch_prepost, $scratchfile_mb,          $safety_margin_mb,   $initially_free);
+    my ($nmax_snapshots,    $number_limit_upper_orig, $number_limit_upper, $n);
+
+    # three test parameters; hardcoded values found appropriate in past tests
+    $n_scratch_prepost = 3;
+    $scratchfile_mb    = 1024;
+    # Expected to keep the test from filling the last 300 MB of the filesystem, poo#36838
+    $safety_margin_mb = 300 + $scratchfile_mb;
+
     assert_script_run("snapper get-config");    # get initial cfg
+    $number_limit_upper_orig = script_output("snapper get-config | awk '/^NUMBER_LIMIT\\>/{ print \$3 }'");
     assert_script_run("snapper list");          # get initial list of snap's
-    assert_script_run("snapper set-config NUMBER_MIN_AGE=0");
+
+    # check amount of free fs disk space and adapt to it
+    $initially_free = get_space("$btrfs_fs_usage | awk -F ' ' '/Free .estimated.:.*min:/{print\$3}'");    # bytes
+    $nmax_snapshots = int(($initially_free / 1_048_576 - $safety_margin_mb) / $scratchfile_mb) * 2;
+
+    # NUMBER_LIMIT setting such that disk space after cleanup
+    # allows another $n_scratch_prepost pre/post snapshot pairs
+    $number_limit_upper = $nmax_snapshots - $n_scratch_prepost * 2;
+    if ($number_limit_upper > 0) {
+        # plenty of free space? No ambition to fill it all, then.
+        $number_limit_upper = 18 if ($number_limit_upper > 18);
+        assert_script_run "snapper set-config NUMBER_LIMIT=2-$number_limit_upper NUMBER_MIN_AGE=0";
+    }
+    elsif ($number_limit_upper == 0) {
+        assert_script_run "snapper set-config NUMBER_LIMIT=0 NUMBER_MIN_AGE=0";
+    }
+    else {
+        # no cleanup test at all will be run in this case, see $n below
+        record_soft_failure('WARNING: insufficient disk space left on / for this test');
```

Comment:
Please use record_soft_failure only with valid bug references. Better to replace `record_soft_failure` with `die` to be explicit. Then after reporting a bug we can still introduce a record_soft_failure referencing that specific bug

Reply:
Point taken, will do. Actually I'd have preferred a real fail here, too, but was unsure how to code it.
Note that the above only happens if the test gets less than about 4.3 GB of initial scratch space, in which case the test would be really pointless. Also, this is not supposed to happen. All the same I wanted to cover this case too, for completeness.

Reply:
To fail, simply write something like:
```perl
die "Not enough space for snapshots, aborting"
```
This will also trigger post_fail_hook to collect logs for investigation.